plugin-server/src/main/runner/on-event.ts

```diff
@@ -0,0 +1,25 @@
+import { PluginEvent } from '@posthog/plugin-scaffold'
+
+import { runInstrumentedFunction } from '../../main/ingestion-queues/ingest-event'
+import { Hub, RunnerWorkerMethods } from '../../types'
+
+export async function onEvent(
+    server: Hub,
+    workerMethods: RunnerWorkerMethods,
+    event: PluginEvent,
+    checkAndPause?: () => void // pause incoming messages if we are slow in getting them out again
```

Comment:
Unrelated to this pr largely, but are we ever using this anywhere?

Reply:
Yup, check `posthog/plugin-server/src/main/ingestion-queues/queue.ts`